components/pagination/pagination.ts

```diff
@@ -0,0 +1,163 @@
+import { Component, Input, Output, OnChanges, SimpleChanges, EventEmitter } from "@angular/core";
+
+@Component({
+    selector: "sui-pagination",
+    template: `
+<div class="ui pagination menu">
+    <a *ngIf="showBoundary" class="item"  (click)="selectPage(1)" [ngClass]="{'disabled': page===1}">
+        <i class="fast backward icon"></i>
+    </a>
+    <a *ngIf="showNavigation" class="item" (click)="selectPage(page-1)" [ngClass]="{'disabled': !hasPrevious()}">
+        <i class="step backward icon"></i>
+    </a>
+    <a *ngFor="let p of pages" class="item" [ngClass]="{'active':p==page}" (click)="page=p">
+        {{ p }}
+    </a>
+    <a *ngIf="showNavigation" class="item" (click)="selectPage(page+1)" [ngClass]="{'disabled': !hasNext()}">
+        <i class="step forward icon"></i>
+    </a>
+    <a *ngIf="showBoundary" class="item"  (click)="selectPage(pageCount)" [ngClass]="{'disabled': page===pageCount}">
+        <i class="fast forward icon"></i>
+    </a>
+</div>
+`
+})
+export class SuiPagination implements OnChanges {
+
+    // Public members
+    public pages:number[] = [];
```

Comment:
Slightly less performant I know but slightly easier to manage (don't have to keep updating the pages array), use an accessor similar to the rating component that creates an array of the right size when needed:

```typescript
public get pages():number[] {
    const [start, end] = this.applyPagination();

    return Array<number>(end - start)
        .fill(start + 1)
        .map((s, i) => s + i);
}
```